src/model/priors/single_cell_bulk.rs

```diff
@@ -142,14 +142,12 @@ impl PairModel<DiscreteAlleleFreqs, ContinuousAlleleFreqs> for SingleCellBulkMod
         };
         let k_single = 0..n_obs_s + 1;
 
-        let k_start = if *af_bulk.start == 0.0 { // 0 as a bulk range start is always inclusive
-            0 as u64
-        } else {
-            // any non-zero bulk range start is always exclusive
-            (*af_bulk.start * n_obs_bulk as f64).floor() as u64 + 1
-        };
-        // any bulk range end is always inclusive
-        let k_end = (*af_bulk.end * n_obs_bulk as f64).floor() as u64 + 1;
+        // stable rust ranges currently only allow inclusive start points, for exclusive add
+        // 0.00000001 in the Event definition
+        let k_start = (*af_bulk.start * n_obs_bulk as f64).ceil() as u64;
+        // stable rust ranges currently only allow exclusive end points, for including an end point,
+        // add 0.00000001 in the Event definition
+        let k_end = (*af_bulk.end * n_obs_bulk as f64).floor() as u64;
```

Comment:
Ok, I understand the issue now. Would the following work?

```
/// An allele frequency range (left- and right-inclusive by default).
pub struct AFRange {
    inner: pub Range<AlleleFreq>,
    pub left_exclusive: bool,
    pub right_exclusive: bool
}

impl {
    pub fn left_exclusive(self) -> Self {
        self.left_exclusive = true;
    }
    
    pub fn right_exclusive(self) -> Self {
        self.left_exclusive = true;
    }
}
```

An event could then be instantiated like this:
```
libprosic::call::pairwise::PairEvent {
            name: "germline_hom_ref".to_owned(),
            af_case: vec![AlleleFreq(0.0)],
            af_control: AFRange(AlleleFreq(0.0)..AlleleFreq(0.25)).right_exclusive()
}
```